bundle/Command/ToggleContentVisibilityCommand.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\Bundle\IbexaScheduledVisibilityBundle\Command;
+
+use Ibexa\Contracts\Core\Repository\Repository;
+use Ibexa\Contracts\Core\Repository\Values\Content\Query;
+use Ibexa\Contracts\Core\Repository\Values\Content\Query\Criterion;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Service\ScheduledVisibilityService;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+use function count;
+
+final class ToggleContentVisibilityCommand extends Command
+{
+    private SymfonyStyle $style;
+
+    public function __construct(
+        private readonly Repository $repository,
+        private readonly ScheduledVisibilityService $scheduledVisibilityService,
+        private readonly bool $enabled,
+        private readonly bool $allContentTypes,
+        private readonly array $allowedContentTypes,
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this->setDescription(
+            'Toggles content visibility based on publish_from and publish_to attributes and configuration.',
+        );
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output): void
+    {
+        $this->style = new SymfonyStyle($input, $output);
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $this->style->info(
+            'This command fetches content and toggles visibility based on its schedule from publish_from and publish_to fields.',
+        );
+
+        if (!$this->enabled) {
+            $this->style->warning('Scheduled visibility mechanism is disabled.');
+
+            return Command::FAILURE;
+        }
+
+        $query = new Query();
+        $query->limit = 50;
+        $criteria = new Criterion\LogicalOr(
+            [
+                new Criterion\IsFieldEmpty('publish_from', false),
+                new Criterion\IsFieldEmpty('publish_to', false),
+            ],
+        );
+
+        if (!$this->allContentTypes) {
```

Comment:
Always try to minimize number of the execution paths, this can be rewritten:

```php
if (!$this->allContentTypes && count($this->allowedContentTypes) === 0) {
    $this->style->warning('No content types configured for scheduled visibility mechanism.');

    return Command::FAILURE;
}

if (count($this->allowedContentTypes) > 0) {
    $criteria = new Criterion\LogicalAnd(
        [
            $criteria,
            new Criterion\ContentTypeIdentifier($this->allowedContentTypes),
        ],
    );
}
```

Also do check https://en.wikipedia.org/wiki/Cyclomatic_complexity.

Reply:
I understand but if allContentTypes option is enabled and allowedContentTypes is filled in, this would overwrite the allContentTypes option and it wouldn't use all of them. I am not sure if that should work like that.

Reply:
I see, then maybe like this:

```php
if (!$this->allContentTypes && count($this->allowedContentTypes) === 0) {
    $this->style->warning('No content types configured for scheduled visibility mechanism.');

    return Command::FAILURE;
}

if (!$this->allContentTypes && count($this->allowedContentTypes) > 0) {
    $criteria = new Criterion\LogicalAnd(
        [
            $criteria,
            new Criterion\ContentTypeIdentifier($this->allowedContentTypes),
        ],
    );
}
```

Reply:
fixed in a53cf31ef5c62a5fa4c19873a5e5a089cfed0e0e